### PR TITLE
Remove Utility.IsMono, related Mono-specific Code 

### LINKED ISSUE #4525: Replace usages of Utility.IsMono
We should remove usages of `Utility.IsMono`, as it will always return `false` in .NET 5.  The usages need to be studied to see if separate code paths are still necessary (most likely not, but there may be some usages that were proxies for os-specific code).

### PR DESCRIPTION
**Pull request by drehtisch**
I removed `Utility.IsMono`, `Utility.MonoVersion`, `Utility.MonoDisplayVersion` and everything that referenced those properties. 

Also removed a bool flag `USE_HTTP_CLIENT_DEFAULT` because of if it always being true after the removal of `Utility.IsMono`

Resolves #4525 

## Remarks

It might be a idea instead of removing `%MONO%` from help.txt, to replace it with `dotnet`. Which should work for all platforms.

### CODE REVIEW

**Comment by warwickmm** [comment]
Thanks @drehtisch.  Does this compile for you?  There appears to be a usage of `USE_HTTP_CLIENT_DEFAULT` remaining here:

https://github.com/drehtisch/duplicati/blob/be658da277185d7e90bbfe24de3eb669d897cd7c/Duplicati/Library/Backend/OneDrive/MicrosoftGraphBackend.cs#L184

Also, there is JavaScript code that references mono.  Here's one example (there may be more):

https://github.com/drehtisch/duplicati/blob/be658da277185d7e90bbfe24de3eb669d897cd7c/Duplicati/Server/webroot/ngax/scripts/directives/backupEditUri.js#L147

Would you be able to remove these references as well?

**Comment by drehtisch** [comment]
Thanks for your feedback @warwickmm. Sorry, I probably missed that. I will remove these references of course. And I will also look again for more mono references.

**Comment by drehtisch** [comment]
While searching the solution for more mono specific code, some questions came up for me. I would like to hear your opinion / feedback on these. 

1. I suppose "__MonoCS__" should be removed in another pull request, because of it being object of another issue (#4524) or should i include that here ?
2. What do we do with the dependency `MonoMac`, see `GUI.TrayIcon.Program:TryGetMonoMac()`, remove it ?
3. Do we still need `Library.Utility.Strings.SslCertificateValidator.MonoHelpSSL`?. It might still be relevant under linux. If that is the case the message needs to be reworded and the property renamed.

**Comment by warwickmm** [comment]
> 1. I suppose __MonoCS__ should be removed in another pull request, because of it being object of another issue (#4524) or should i include that here ?

I think it's fine to remove `__MonoCS__` in this pull request.

> 2. What do we do with the dependency MonoMac, see GUI.TrayIcon.Program:TryGetMonoMac(), remove it ?

Based on [this comment](https://github.com/duplicati/duplicati/pull/3124#issuecomment-436915441), it sounds like we can remove MonoMac.

> 3. Do we still need Library.Utility.Strings.SslCertificateValidator.MonoHelpSSL?. It might still be relevant under linux. If that is the case the message needs to be reworded and the property renamed.

I think it would be ok to remove.

Perhaps this pull request should just try and remove as much mono as we can, and then we can get  some other opinions on what parts need to stay, if any.

/cc @mnaiman, @tsuckow, @kenkendk

File: Duplicati/GUI/Duplicati.GUI.TrayIcon/Program.cs [hunk 1]
```diff
@@ -319,14 +314,6 @@ private static TrayIconBase RunTrayIcon(string toolkit)
 
         [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
         private static TrayIconBase GetWinformsInstance() { return new Win32Runner(); }
-#if __MonoCS__ || __WindowsGTK__ || ENABLE_GTK
```
Comment (warwickmm) [inline Duplicati/GUI/Duplicati.GUI.TrayIcon/Program.cs:322]: I think we only want to remove the `__MonoCS__` part of the `#if` condition.  The rest of this code should remain.

Comment (drehtisch) [reply Duplicati/GUI/Duplicati.GUI.TrayIcon/Program.cs:322]: Alright. Will restore that then.

File: Duplicati/Library/SQLiteHelper/SQLiteLoader.cs [hunk 2]
```diff
@@ -165,67 +165,34 @@ public static Type SQLiteConnectionType
...
 
+                // If we have a new path, try to force load the mixed-mode assembly for the current architecture
+                // This can be avoided if the preload in SQLite works, but it is easy to do it here as well
+                if (loadMixedModeAssembly)
+                {
+                    try { PInvoke.LoadLibraryEx(Path.Combine(assemblyPath, "SQLite.Interop.dll"), IntPtr.Zero, 0); }
+                    catch (Exception ex) { Logging.Log.WriteExplicitMessage(LOGTAG, "LoadMixedModeSQLiteError", ex, "Failed to load the mixed mode SQLite database: {0}", Path.Combine(assemblyPath, "SQLite.Interop.dll")); }
+                }
+
+
```
Comment (warwickmm) [inline Duplicati/Library/SQLiteHelper/SQLiteLoader.cs:195]: This will be cleaned up further when we address issue #4526.

**Comment by warwickmm** [comment]
This looks good to me.  Thanks @drehtisch.

**Pull request merged**